### PR TITLE
test: Ensure access to reduced counterexamples from fast-check

### PR DESCRIPTION
**Pull request by gibson042**
## Description

Updates use of fast-check assert to conform with its documentation: https://dubzzz.github.io/fast-check/index.html#assert

> WARNING: Has to be awaited
> …
> Returns Promise\<void>

### Security Considerations

n/a

### Documentation Considerations

n/a

### Testing Considerations

Noted above.

### CODE REVIEW

**Review by Chris-Hibbert — APPROVED** [review]
Thanks for noticing this, Richard.

FYI, @dckc, who introduced these tests.

**Comment by dckc** [comment]
Weird. Their [tutorial example](https://github.com/dubzzz/fast-check/blob/main/documentation/HandsOnPropertyBased.md#hands-on-fast-check) treats `fc.assert` as sync.

**Comment by gibson042** [comment]
It took some digging, but I think I got to the bottom of it—`fc.assert` has multiple signatures, and is asynchronous if and only if the property being tested is asynchronous. So landing this PR is safe (and possibly future-proof) but unnecessary.

**Comment by gibson042** [comment]
I also opened an issue in fast-check: https://github.com/dubzzz/fast-check/issues/2749

File: packages/ERTP/test/unitTests/test-amountProperties.js [hunk 1]
```diff
@@ -28,8 +28,8 @@ const arbAmount = arbBagContents.map(contents =>
 // Note: we write P => Q as !P || Q since JS has no logical => operator
 const implies = (p, q) => !p || q;
 
-test('isEqual is a (total) equivalence relation', t => {
-  fc.assert(
+test('isEqual is a (total) equivalence relation', async t => {
+  await fc.assert(
```
Comment (dubzzz) [inline packages/ERTP/test/unitTests/test-amountProperties.js:33]: Not needed given the fact that you are using a sync property.

File: packages/ERTP/test/unitTests/test-amountProperties.js [hunk 2]
```diff
@@ -43,9 +43,9 @@ test('isEqual is a (total) equivalence relation', t => {
   );
 });
 
-test('isGTE is a partial order with empty as minimum', t => {
+test('isGTE is a partial order with empty as minimum', async t => {
   const empty = m.makeEmpty(mockBrand, AssetKind.COPY_BAG);
-  fc.assert(
+  await fc.assert(
```
Comment (dubzzz) [inline packages/ERTP/test/unitTests/test-amountProperties.js:64]: Same

File: packages/ERTP/test/unitTests/test-amountProperties.js [hunk 3]
```diff
@@ -56,9 +56,9 @@ test('isGTE is a partial order with empty as minimum', t => {
   );
 });
 
-test('add: closed, commutative, associative, monotonic, with empty identity', t => {
+test('add: closed, commutative, associative, monotonic, with empty identity', async t => {
   const empty = m.makeEmpty(mockBrand, AssetKind.COPY_BAG);
-  fc.assert(
+  await fc.assert(
```
Comment (dubzzz) [inline packages/ERTP/test/unitTests/test-amountProperties.js:86]: Same

File: packages/ERTP/test/unitTests/test-amountProperties.js [hunk 4]
```diff
@@ -76,8 +76,8 @@ test('add: closed, commutative, associative, monotonic, with empty identity', t
   );
 });
 
-test('subtract: (x + y) - y = x; (y - x) + x = y if y >= x', t => {
-  fc.assert(
+test('subtract: (x + y) - y = x; (y - x) + x = y if y >= x', async t => {
+  await fc.assert(
```
Comment (dubzzz) [inline packages/ERTP/test/unitTests/test-amountProperties.js:127]: Same

File: packages/run-protocol/test/amm/vpool-xyk-amm/test-amm-calc.js [hunk 5]
```diff
@@ -37,8 +37,8 @@ const withinEpsilon = (left, right) =>
   multiply(right, 105) >= multiply(left, 100) &&
   multiply(left, 105) >= multiply(right, 100);
 
-test('balancesToReachRatio calculations are to spec', t => {
-  fc.assert(
+test('balancesToReachRatio calculations are to spec', async t => {
+  await fc.assert(
```
Comment (dubzzz) [inline packages/run-protocol/test/amm/vpool-xyk-amm/test-amm-calc.js:41]: Same

**Comment by dubzzz** [comment]
Only `asyncProperty` has to be awaited. Normal `property` can be but it is not needed. Anyway, always awaiting can avoid forgetting about it and will work for both (while not awaiting the async one will miss bugs).

**Pull request merged**